src/sentry/reprocessing2.py

```diff
@@ -347,7 +350,96 @@ def _get_info_reprocessed_key(group_id):
     return f"re2:info:{group_id}"
 
 
-def mark_event_reprocessed(data=None, group_id=None, project_id=None):
+def buffered_handle_remaining_events(
+    project_id: int,
+    old_group_id: int,
+    new_group_id: int,
+    datetime_to_event: List[Tuple[datetime, str]],
+    remaining_events,
+    force_flush_batch: bool = False,
+):
+    """
+    A quick-and-dirty wrapper around `handle_remaining_events` that batches up
+    event IDs in Redis. We need this because Snuba cannot handle many tiny
+    messages and prefers big ones instead.
+
+    For optimal performance, the datetimes should be close to each other. This
+    "soft" precondition is fulfilled in `reprocess_group` by iterating through
+    events in timestamp order.
+
+    Ideally we'd have batching implemented via a service like buffers, but for
+    more than counters.
+    """
+
+    client = _get_sync_redis_client()
+    # We explicitly cluster by only project_id and group_id here such that our
+    # RENAME command later succeeds.
+    #
+    # We also use legacy string formatting here because new-style Python
+    # formatting is quite confusing when the output string is supposed to
+    # contain {}.
+    key = f"re2:remaining:{{{project_id}:{old_group_id}}}"
+
+    llen = None
+
+    if datetime_to_event:
+        llen = client.lpush(
+            key,
+            *(f"{to_timestamp(datetime)};{event_id}" for datetime, event_id in datetime_to_event),
+        )
+        client.expire(key, settings.SENTRY_REPROCESSING_SYNC_TTL)
+
+    MAX_LLEN = settings.SENTRY_REPROCESSING_REMAINING_EVENTS_BUF_SIZE
+
+    if force_flush_batch or (llen is not None and llen > MAX_LLEN) or client.llen(key) > MAX_LLEN:
+        new_key = f"{key}:{uuid.uuid4().hex}"
+
+        try:
+            # Rename `key` to a new temp key that is passed to celery task. We
+            # use `renamenx` instead of `rename` only to detect UUID collisions.
+            assert client.renamenx(key, new_key), "UUID collision for new_key?"
```

Comment:
Optional: Can we catch this down and report to Sentry?

Reply:
I think we should halt reprocessing... if we simply continue with the next page, `key` might become too large.

Reply:
we could potentially catch down the assertion, delete `key` from redis and sort-of chug on with reprocessing

but I think the best way to recover from random errors in reprocessing could be: abort reprocessing, merge the old and new issues together via regular issue merge, tell the user "sorry" and hope no data was lost